src/trigger.rs

```diff
@@ -969,114 +974,101 @@ pub fn validate_target_ok_this(
-                old_warn(part, ErrorKey::Validation, &msg);
             }
-            if part.lowercase_is("root") {
-                sc.replace_root();
-            } else if part.lowercase_is("prev") {
-                sc.replace_prev();
-            } else {
-                sc.replace_this();
-            }
-        } else if let Some((inscopes, outscope)) = scope_to_scope(part, sc.scopes()) {
-            if inscopes == Scopes::None && !first {
-                let msg = format!("`{part}` makes no sense except as first part");
-                old_warn(part, ErrorKey::Validation, &msg);
-            }
-            sc.expect(inscopes, &Reason::Token(part.clone()));
-            sc.replace(outscope, part.clone());
-        } else if data.script_values.exists(part.as_str()) {
-            data.script_values.validate_call(part, data, sc);
-            sc.replace(Scopes::Value, part.clone());
-        } else if let Some(inscopes) = trigger_comparevalue(part, data) {
-            if !last {
-                let msg = format!("`{part}` only makes sense as the last part");
-                old_warn(part, ErrorKey::Scopes, &msg);
-                sc.close();
-                return;
-            }
-            if inscopes == Scopes::None && !first {
-                let msg = format!("`{part}` makes no sense except as first part");
-                old_warn(part, ErrorKey::Validation, &msg);
-            }
-            if part.is("current_year") && sc.scopes() == Scopes::None {
-                warn_info(
-                    part,
-                    ErrorKey::Bugs,
-                    "current_year does not work in empty scope",
-                    "try using current_date, or dummy_male.current_year",
-                );
-            } else {
-                sc.expect(inscopes, &Reason::Token(part.clone()));
-            }
-            sc.replace(Scopes::Value, part.clone());
-        } else {
-            // The part is not found. Issue an appropriate warning.
-            // TODO: warn if trying to use iterator here
-
-            // See if the user forgot a prefix like `faith:` or `cuture:`
-            let mut opt_info = None;
-            if first && last {
-                if let Some(prefix) = needs_prefix(part.as_str(), data, outscopes) {
-                    opt_info = Some(format!("did you mean `{prefix}:{part}` ?"));
-                }
-            };
-
-            let msg = format!("unknown token `{part}`");
-            err(ErrorKey::UnknownField).msg(msg).opt_info(opt_info).loc(part).push();
-            sc.close();
-            return;
         }
     }
     let (final_scopes, because) = sc.scopes_reason();
     if !outscopes.intersects(final_scopes | Scopes::None) {
         let part = &part_vec[part_vec.len() - 1];
         let msg = format!("`{part}` produces {final_scopes} but expected {outscopes}");
-        if part == because.token() && part.loc == because.token().loc {
```

Comment:
I remember putting this in because otherwise it led to spurious extra message blocks that pointed to the same token as the original block